### PR TITLE
[SHELL32] Improve empty recycle bin sound code

### PR DESCRIPTION
**Pull request by TAN-Gaming**
## Purpose
- Split off from [#4755](https://github.com/reactos/reactos/pull/4755) (This change doesn't require [#4635](https://github.com/reactos/reactos/pull/4635) to work.)

## Proposed changes
- Remove `MAX_PATH` limit
- Use ATL classes
- Play the sound asynchronously

### CODE REVIEW

**Review by learn-more — APPROVED** [review]

**Review by binarymaster — APPROVED** [review]

File: dll/win32/shell32/folders/CRecycleBin.cpp [hunk 1]
```diff
@@ -1018,6 +1018,57 @@ TRASH_TrashFile(LPCWSTR wszPath)
...
+        /* The type is REG_SZ, no need to expand */
+        pszSndPath.Attach(pszValue.Detach());
+    }
+    else
+    {
+        /* Invalid type */
+        return;
+    }
+
+    PlaySoundW(pszSndPath, NULL, SND_FILENAME | SND_ASYNC | SND_NODEFAULT);
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CRecycleBin.cpp:1069]: Question: Shouldn't this be ideally, just one simple call to `PlaySoundW(L"EmptyRecycleBin", NULL, SND_ALIAS | SND_APPLICATION);` , the PlaySoundW function, thanks to the SND_APPLICATION, doing all the lifting work of opening the correct registry key based on the current name of the application?

**Review by HBelusca — CHANGES_REQUESTED** [review]
Some stuff to investigate...

File: dll/win32/shell32/folders/CRecycleBin.cpp (hunk 1 above)
Comment (TAN-Gaming) [reply dll/win32/shell32/folders/CRecycleBin.cpp:1069]: Note: `SND_APPLICATION` relies on `GetModuleFileName` which will break if you changed the file name from explorer.exe to something else.

Comment (TAN-Gaming) [reply dll/win32/shell32/folders/CRecycleBin.cpp:1069]: PR [#4635](https://github.com/reactos/reactos/pull/4635) is required for `SND_APPLICATION` to work too.

Comment (TAN-Gaming) [reply dll/win32/shell32/folders/CRecycleBin.cpp:1069]: I tried calling `SHEmptyRecycleBinW(NULL, NULL, SHERB_NOCONFIRMATION | SHERB_NOPROGRESSUI);` on Windows 7 from the test application named `Test_RecycleBin.exe` and created the following registry key:
Capture_Win7

The function still uses the sound from Explorer so `SND_APPLICATION` flag is not used.

**Review by HBelusca — APPROVED** [review]

**Pull request merged**